pkg/apis/etcd/v1beta2/status.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta2
+
+import (
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type ClusterPhase string
+type ClusterConditionType string
+
+const (
+	ClusterPhaseNone     ClusterPhase = ""
+	ClusterPhaseCreating              = "Creating"
+	ClusterPhaseRunning               = "Running"
+	ClusterPhaseFailed                = "Failed"
+
+	// See ./doc/user/conditions_and_events.md
+	ClusterConditionAvailable  ClusterConditionType = "Available"
+	ClusterConditionRecovering                      = "Recovering"
+	ClusterConditionScaling                         = "Scaling"
+	ClusterConditionUpgrading                       = "Upgrading"
+)
+
+type ClusterStatus struct {
+	// Phase is the cluster running phase
+	Phase  ClusterPhase `json:"phase"`
+	Reason string       `json:"reason"`
+
+	// ControlPuased indicates the operator pauses the control of the cluster.
+	ControlPaused bool `json:"controlPaused"`
+
+	// Condition keeps ten most recent cluster conditions
```

Comment:
Condition keeps track of all cluster conditions, if they exist.

Reply:
fixed